openff/toolkit/tests/test_topology.py

```diff
@@ -1271,3 +1271,62 @@ def test_tagsorted_dict_clear(tsd):
     tsd.clear()
 
     assert len(tsd) == 0
+
+
+class TestValenceDict:
+    @pytest.mark.parametrize(
+        "inkey, outkey",
+        [
+            [[0, 1], (0, 1)],
+            [[1, 0], (0, 1)],
+            [(1, 3, 2, 4), (1, 3, 2, 4)],
+            [(4, 2, 3, 1), (1, 3, 2, 4)],
+            [(1, 8, 3, 4, 2, 5), (1, 8, 3, 4, 2, 5)],
+            [(5, 2, 4, 3, 8, 1), (1, 8, 3, 4, 2, 5)],
+        ],
+    )
+    def test_key_transform(self, inkey, outkey):
+        assert ValenceDict.key_transform(inkey) == outkey
+
+    @pytest.mark.parametrize(
+        "key, possible, index",
+        [
+            [[0, 1], None, 0],
+            [[1, 0], None, 1],
+            [(1, 3, 2, 4), None, 0],
+            [(4, 2, 3, 1), None, 1],
+            [
+                (4, 2, 3, 1),
+                [
+                    (4, 2, 3, 1),
+                ],
+                0,
```

Comment:
This is how black wanted it -- I personally would not keep the code formatted like this, but I'll leave it to y'all

Reply:
:shakes fist at uncompromising formatter: "You were supposed to bring peace!"

Yeah, that's ugly. Ah well, I'll take the good with the bad with `black`

Reply:
Also not but one can split out the arguments into multiple decorators and the function will gobble everything up


```
import pytest

@pytest.mark.parametrize("x", [(1, 2)])
@pytest.mark.parametrize("y", [(1, 2)])
def test_foo(x, y):
    assert x == y

#     test_foo.py .                                                            [100%]
# 
# ============================== 1 passed in 0.06s ===============================